### PR TITLE
copytranslator: Add version 9.0.0

### PR DESCRIPTION
**Pull request by davidxuang**
[CopyTranslator](https://github.com/CopyTranslator/CopyTranslator)

### CODE REVIEW

File: bucket/copytranslator.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v9.0.0/copytranslator-setup-9.0.0.exe#dl.7z",
+            "hash": "sha512:d8e6f38027ede2692b3ff2242d970923c8c468ac07bc231bcf169d6e18b947d1edb778d4256a0f64ebfed97d8500e70799d453e94c58e8176ec43228cf0dc15a",
+            "installer": {
+                "script": "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\""
+            }
+        }
+    },
+    "post_install": [
```
Comment (Ash258) [inline bucket/copytranslator.json]:
```suggestion
    "post_install": "Remove-Item \"$dir\\`$PLUGINSDIR\", \"$dir\\Uninstall*\" -Recurse",
```

File: bucket/copytranslator.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    ],
+    "checkver": {
+        "github": "https://github.com/copytranslator/copytranslator"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v$version/copytranslator-setup-$version.exe",
+                "hash": {
+                    "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v$version/latest.yml",
```
Comment (Ash258) [inline bucket/copytranslator.json]:
```suggestion
                    "url": "$baseurl/latest.yml",
```

File: bucket/copytranslator.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                    "regex": "([a-zA-Z0-9+/=]{88})"
```
Comment (Ash258) [inline bucket/copytranslator.json]:
```suggestion
                    "regex": "sha512:\\s+$base64"
```

File: bucket/copytranslator.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+    "version": "9.0.0",
+    "description": "Foreign language translating assistant basing on clipboard.",
```
Comment (Ash258) [inline bucket/copytranslator.json]:
```suggestion
    "description": "Foreign language assisted reader and translator",
```

**Comment by davidxuang** [comment]
> Application without proper homepage and documentation page in English cannot be accepted into official bucket.

Changed the homepage to README in English already. Is it detailed enough or I should contact the author for more translated documents?

File: bucket/copytranslator.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        [
+            "copytranslator.exe",
+            "Copy Translator"
+        ]
+    ],
+    "checkver": "github",
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v$version/copytranslator-setup-$version.exe",
```
Comment (Ash258) [inline bucket/copytranslator.json]:
```suggestion
                "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v$version/copytranslator-setup-$version.exe#/dl.7z",
```

File: bucket/copytranslator.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": "9.0.0",
+    "description": "Foreign language assisted reader and translator",
+    "homepage": "https://github.com/copytranslator/copytranslator",
+    "license": "GPL-2.0-only",
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v9.0.0/copytranslator-setup-9.0.0.exe#dl.7z",
```
Comment (Ash258) [inline bucket/copytranslator.json]:
```suggestion
            "url": "https://github.com/CopyTranslator/CopyTranslator/releases/download/v9.0.0/copytranslator-setup-9.0.0.exe#/dl.7z",
```

**Comment by Ash258** [comment]
- No option to enable gui in English
- No configuration option for English

I though it is obvious that application without English support cannot be in official bucket, but even when I mentioned it you forgot the small problem that application do not support English in GUI and everywhere and there is no option to have it...

**Comment by davidxuang** [comment]
This is really weird… I saw the locale files in the source repo so I ignored this, will check it back later and see if there's a way to overwrite config while installing.

**Comment by linsui** [comment]
Just set locale as English.
sshot-002

**Comment by Ash258** [comment]
If you can configure it using config file, then I will reopen. But if user have to first start up application and then configure within chinese GUI, it is unacceptable

**Comment by linsui** [comment]
It's possible. Somethinf like this may work.

```
    "post_install": [
        "Remove-Item \"$dir\\`$PLUGINSDIR\", \"$dir\\Uninstall*\" -Recurse",
        "if (!(Test-Path \"~\\copytranslator\\copytranslator.json\")) {",
        "    New-Item \"~\\copytranslator\\copytranslator.json\" -Value \"{`\"version`\": `\"$version`\", `\"localeSetting`\": `\"en`\"}\" -Force | Out-Null",
        "}"
    ],
```

However the `version` in 9.0.0 is `"version": "v9.0.0 寒渐 ",` but `"version": "9.0.1",`. Without correct version in this file all settings are override. After update the version will change and it seems all settings are override. I do not know why. I suuggest to submit it to dorado of contact upstream. @davidxuang

**Comment by davidxuang** [comment]
This script should be working. ([The commit](https://github.com/davidxuang/scoop-extras/commit/f811191bfbbcb8a7b025cbcd7f656e7afb15d1af)) @Ash258 

```json
    "post_install": [
        "Remove-Item \"$dir\\`$PLUGINSDIR\", \"$dir\\Uninstall*\" -Recurse",
        "if ((Get-WinSystemLocale)[0].ToString().SubString(0,2) -ne \"zh\") {",
        "    Start-Process $dir\\copytranslator.exe",
        "    Start-Sleep 2",
        "    Get-Process -Name copytranslator | Stop-Process",
        "    $configPath = \"~\\copytranslator\\copytranslator.json\"",
        "    $config = Get-Content $configPath | ConvertFrom-Json",
        "    $config.localeSetting = \"en\"",
        "    ConvertTo-Json $config | Set-Content $configPath",
        "}"
```

**Comment by davidxuang** [comment]
Maybe I should just wait for the original author to fix the issue: they provides English GUI but do not detect system locale, which makes the localisation meaningless...

**Pull request closed without merging**